overlord/snapstate/backend/setup.go

```diff
@@ -92,8 +100,10 @@ func (b Backend) SetupSnap(snapFilePath, instanceName string, sideInfo *snap.Sid
 	}
 
 	t := s.Type()
-	if err := boot.Kernel(s, t, dev).ExtractKernelAssets(snapf); err != nil {
-		return snapType, nil, fmt.Errorf("cannot install kernel: %s", err)
+	if !setupOpts.SkipKernelExtraction {
+		if err := boot.Kernel(s, t, dev).ExtractKernelAssets(snapf); err != nil {
+			return snapType, nil, fmt.Errorf("cannot install kernel: %s", err)
```

Comment:
I didn't check if `err` can be of some type which we might want to distinguish but using the `%w` verb would make `fmt.Errorf` return an error that wraps `err`. Probably not be useful here at all, just thought to mention it

Reply:
Thanks for raising this. I won't change it here because the existing code is written in this way but it's a good point. We are underusing %w a bit - partly for historic reasons because until recently we were stuck in the go-1.9 stone age :) Fortunately this has changed now!

Reply:
yeah we should probably have a separate PR which goes through and changes everything over to use %w, we have numerous places around the codebase that could benefit from using %w but never could use it previously.

We also have other cruft too like ctrl17.go and ctrl16.go which should be updated/removed too now that we are on a more modern Go version